hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -147,6 +152,25 @@ class EmitFunctionBuilder[F >: Null](
   private[this] var _hconf: SerializableHadoopConfiguration = _
   private[this] var _hfield: ClassFieldRef[SerializableHadoopConfiguration] = _
 
+  private[this] var _mods: ArrayBuilder[(String, Int => AsmFunction3[Region, Array[Byte], Array[Byte], Array[Byte]])] = new ArrayBuilder()
+  private[this] var _sparkField: ClassFieldRef[SparkBackendUtils] = _
```

Comment:
what does a `_sparkField` represent?

Reply:
SparkBackendUtils is the thing that holds all the partition functions and distributes/runs them. `_sparkField` is the field on the compiled function class that is going to hold this object (gets populated by the call to `addSparkBackend` further down)